### DISCUSSION TITLE
[2022/02/22] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#193] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/192 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
## WAI-ARIA 勉強会

[markuplint presents -UIのアクセシビリティを損なわないための基礎と応用-](https://docs.google.com/presentation/d/1dKi_6-LOcMFpKPqdCXmpdsNPUznjz0n2BuVBbcIa0hE/edit#slide=id.p)

WAIｰARIA の原則がまとまっていて良い。

- WAI-ARIA を使うとHTMLの役割・性質・状態を「一部」拡張できる
  - WAI-ARIA には、指定できる要素と指定できる組み合わせに制限がある
- **ロールがあってプロパティとステートがある**
  - `<button aria-label="ヘルプ">?</button>`
    - 🙅‍♂️ 「button要素のaria-label属性を指定している」
    - 🙆‍♀️ 「buttonロールのaria-labelプロパティを設定している」
  - **HTML要素は暗黙のロールを持つ** ← ここ結構知らない人いそう（自分もちゃんとテストを書くようになるまで知らなかった）
- HTMLが持っているロールをなるべく使う
- ... （他にもルールがたくさんある）

**Reply by kobayang** [reply]
https://coliss.com/articles/build-websites/operation/html-attributes-you-may-want-to-use.html

便利HTML属性一覧。

**Comment by kobayang** [comment]
## The State of JS 2021

結果が出てた。（まだ見てない）
https://2021.stateofjs.com/ja-JP/

**Comment by kobayang** [comment]
https://github.com/solidjs/solid
